pkgs/top-level/all-packages.nix

```
@@ -23041,7 +23041,12 @@ with pkgs;
 
   onlyoffice-documentserver = callPackage ../servers/onlyoffice-documentserver { };
 
-  outline = callPackage ../servers/web-apps/outline { };
+  outline = callPackage ../servers/web-apps/outline (
+  let overrides = (super: {
+    yarn2nix-moretea = yarn2nix-moretea.override { inherit (super) nodejs yarn; };
+    yarn = yarn.override { inherit (super) nodejs; };
+    nodejs = nodejs-16_x;
+  }) overrides; in overrides);
```

Isn't this basically the same as

```nix
lib.fix (super: {
  yarn2nix-moretea = yarn2nix-moretea.override { inherit (super) nodejs yarn; };
  yarn = yarn.override { inherit (super) nodejs; };
  nodejs = nodejs-16_x;
})
```

which should evaluate to something like

```nix
rec {
  nodejs = nodejs-16_x;
  yarn = yarn.override { inherit nodejs; };
  yarn2nix-moretea = yarn2nix-moretea.override { inherit nodejs yarn; };
}
```

If I'm not missing something here, ``outline.override { nodejs = someOtherNodejs };`` would only override the Node interpreter and won't propagate the overriden Node to Yarn or yarn2nix. So what's the point in defining a complex overlay here when this would do the same thing:

```suggestion
  outline = callPackage ../servers/web-apps/outline (rec {
    nodejs = nodejs-16_x;
    yarn = yarn.override { inherit nodejs; };
    yarn2nix-moretea = yarn2nix-moretea.override { inherit nodejs yarn; };
  });
```

---

It won't work with `rec` — yarn will cause an infinite recursion

---

However, `lib.fix` really does pretty much the same thing I did, yeah)

---

I see – then I vote for `lib.fix` ^^

---

god setting correct co-author without a public email is hard D: